src/utils/Draft.js

```diff
@@ -71,66 +78,123 @@ function botRating(botColors, card) {
 }
 
+async function buildDeck(cards, bot) {
+  const nonlands = cards.filter((card) => !card.type_line.toLowerCase().includes('land'));
+  const lands = cards.filter((card) => card.type_line.toLowerCase().includes('land'));
+
+  const sort_fn = function(a, b) {
+    if (bot) {
+      return botRating(bot, b) - botRating(bot, a);
+    } else {
+      return (draft.ratings[b.details.name] = draft.ratings[a.details.name]);
+    }
+  };
+
+  nonlands.sort(sort_fn);
+  lands.sort(sort_fn);
+
+  const main = nonlands.slice(0, 23).concat(lands.slice(0, 17));
+  const side = nonlands.slice(23).concat(lands.slice(17));
+
+  const deck = [];
+  const sideboard = [];
+  for (let i = 0; i < 16; i += 1) {
+    deck.push([]);
+    if (i < 8) {
+      sideboard.push([]);
+    }
+  }
+
+  for (const card of main) {
+    let index = Math.min(card.cmc, 7);
+    if (!card.type_line.toLowerCase().includes('creature')) {
+      index += 8;
+    }
+    deck[index].push(card);
+  }
+  for (const card of side) {
+    sideboard[Math.min(card.cmc, 7)].push(card);
+  }
+
+  return {
+    deck,
+    sideboard,
+  };
+}
+
 function botPicks() {
   // make bots take one pick out of active packs
-  for (let botIndex = 1; botIndex < draft.packs.length; botIndex++) {
-    const botPack = draft.packs[botIndex][0];
-    const botColors = draft.bots[botIndex - 1];
+  for (let botIndex = 1; botIndex < draft.seats.length; botIndex++) {
+    const pack = draft.seats[botIndex].packbacklog[0];
+    const botColors = draft.seats[botIndex].bot;
     const ratedPicks = [];
     const unratedPicks = [];
-    for (let cardIndex = 0; cardIndex < botPack.length; cardIndex++) {
-      if (draft.ratings[botPack[cardIndex].details.name]) {
+    for (let cardIndex = 0; cardIndex < pack.length; cardIndex++) {
+      if (draft.ratings && draft.ratings[pack[cardIndex].details.name]) {
         ratedPicks.push(cardIndex);
       } else {
         unratedPicks.push(cardIndex);
       }
     }
 
     ratedPicks.sort((x, y) => {
-      return botRating(botColors, botPack[y]) - botRating(botColors, botPack[x]);
+      return -botRating(botColors, pack[x]);
     });
     arrayShuffle(unratedPicks);
 
     const pickOrder = ratedPicks.concat(unratedPicks);
-    const [botPick] = botPack.splice(pickOrder[0], 1);
-    draft.picks[botIndex].push(botPick.cardID);
+    pick = draft.seats[botIndex].packbacklog[0].splice(pickOrder[0], 1)[0];
+    draft.seats[botIndex].pickorder.push(pick);
   }
 }
 
 function passPack() {
-  draft.pickNumber += 1;
   botPicks();
-  // check if pack is done
-  if (draft.packs.every((seat) => seat[0].length === 0)) {
-    draft.packNumber += 1;
-    draft.pickNumber = 1;
-    // splice the first pack out
-    for (const drafter of draft.packs) {
-      drafter.splice(0, 1);
+  //check if pack is done
+  if (draft.seats.every((seat) => seat.packbacklog[0].length === 0)) {
+    //splice the first pack out
+    for (const seat of draft.seats) {
+      seat.packbacklog.splice(0, 1)[0];
```

Comment:
Not setting anything = this? Shouldn't need `[0]` if you're just removing it?

Reply:
correct